ui/src/components/Form.jsx

```diff
@@ -0,0 +1,136 @@
+import React, { useState } from "react";
+import SendingInfo from "./SendingInfo";
+import Header from "./Header";
+import Footer from "./Footer";
+import DocumentSize from "./DocumentSize";
+import { Container, LinearProgress } from "@material-ui/core";
+import "../styles/Form.css";
+import ParcelSize from "./ParcelSize";
+import ParcelType from "./ParcelType";
+import { withSnackbar } from "notistack";
+import Summary from "./Summary";
+
+function Form(props) {
+  var boxSizes = [
+    { name: "Small", width: 35, height: 16, length: 45, weigth: 2 },
+    { name: "Medium", width: 46, height: 46, length: 64, weigth: 20 },
+    { name: "Large", width: 150, height: 150, length: 150, weigth: 30 },
+  ];
+
+  var letterSizes = [
+    { name: "Small Letter", weight: 100, length: 24, width: 16 },
+    { name: "Large Letter", weight: 750, length: 35, width: 25 },
+  ];
+
+  const [selectedPackageType, setSelectedPackageType] = useState(undefined);
+  const [selectedBoxSize, setSelectedBoxSize] = useState(undefined);
+  const [selectedDocumentSize, setSelectedDocumentSize] = useState(undefined);
+  const [selectedPaymentType, setSelectedPaymentType] = useState(undefined);
+  const [currentPage, setCurrentPage] = useState(0);
+  const [progress, setProgress] = useState(25);
+
+  function NextPage(e) {
+    if (currentPage === 0 && selectedPackageType === undefined) {
+      displayError("Parcel type is not selected!", "error");
+      return;
+    } else if (currentPage === 1 && selectedBoxSize === undefined && selectedPackageType === "Box") {
+      displayError("Box size is not selected!", "error");
+      return;
+    } else if (currentPage === 1 && selectedDocumentSize === undefined && selectedPackageType === "Document") {
+      displayError("Letter size is not selected!", "error");
+      return;
+    }
+    setCurrentPage(currentPage + 1);
+    setProgress(progress + 25);
+  }
+
+  function PreviousPage(e) {
+    setCurrentPage(currentPage - 1);
+    setProgress(progress - 25);
+  }
+
+  function displayError(errorMessage, type) {
```

Comment:
To make error messages reusable in other components I would consider moving this function somewhere else. Maybe some wrapper component could work?

Reply:
I can't move it to other component or seperate function as it requires a hook or a prop from parent function to call a snackbar, only solution is to pass the function to children components if you want to call it or use context to call it globally. If you have any better ideas, feel free to commit. [Library documentation](https://iamhosseindhv.com/notistack) if you are interested.